packages/signers/src/web3auth/signer.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  WalletClientSigner,
+  type SignTypedDataParams,
+  type SmartAccountAuthenticator,
+} from "@alchemy/aa-core";
+import { Web3Auth } from "@web3auth/modal";
+import { createWalletClient, custom, type Hash } from "viem";
+import type {
+  Web3AuthAuthenticationParams,
+  Web3AuthUserInfo,
+} from "./types.js";
+
+export class Web3AuthSigner
+  implements
+    SmartAccountAuthenticator<
+      Web3AuthAuthenticationParams,
+      Web3AuthUserInfo,
+      Web3Auth
+    >
+{
+  inner: Web3Auth;
+  private signer: WalletClientSigner | undefined;
+
+  constructor({ inner }: { inner: Web3Auth }) {
+    this.inner = inner;
+  }
+
+  get signerType() {
+    return "web3auth";
+  }
+
+  getAddress = async () => {
+    if (!this.signer) throw new Error("Not authenticated");
+
+    const address = await this.signer.getAddress();
+    if (address == null) throw new Error("No address found");
+
+    return address as Hash;
+  };
+
+  signMessage = async (msg: Uint8Array | string) => {
+    if (!this.signer) throw new Error("Not authenticated");
+
+    return this.signer.signMessage(msg);
+  };
+
+  signTypedData = (params: SignTypedDataParams) => {
+    if (!this.signer) throw new Error("Not authenticated");
+
+    return this.signer.signTypedData(params);
+  };
+
+  authenticate = async (params: Web3AuthAuthenticationParams) => {
+    await params.initModal();
```

Comment:
nit: I think this is usually done when the web3auth sdk is initialized because it will inject the iframe into the page

Reply:
sorry is the recommendation then to remove this and have the developer define it? I'm just copying it from the example: https://accountkit.alchemy.com/smart-accounts/signers/web3auth.html

Reply:
nm yea this is fine